index.js

```diff
@@ -165,3 +205,107 @@ export const Issue = {
   },
   activeLockReason({ source }) { return source['active_lock_reason']; },
 }
+
+export const PullRequestCollection = {
+  async one({ self, source, args }) {
+    const { name: owner } = self.match(root.users.one());
+    const { name: repo } = self.match(root.users.one().repos().one());
+    const { number } = args;
+    const result = await client.pullRequests.get({ owner, repo, number });
+    return result.data;
+  },
+
+  async page({ self, source, args }) {
+    const { name: owner } = self.match(root.users.one());
+    const { name: repo } = self.match(root.users.one().repos().one());
+
+    const apiArgs = toApiArgs(args, { owner, repo });
+    const res = await client.pullRequests.getAll(apiArgs);
+
+    return {
+      items: res.data,
+      next: getNextPageRef(self.page(args), res),
+    };
+  }
+}
+
+export const PullRequest = {
+  self({ self, parent, source }) {
+    return parent.parent.parent.one({ number: source.number });
+  },
+  activeLockReason({ source }) { return source['active_lock_reason']; },
+  files({ self, source }) { return {}; },
+  // async files ({ self, source}){
+  //   const { name: owner } = self.match(root.users.one());
+  //   const { name: repo } = self.match(root.users.one().repos().one());
+  //   const { number } = source;
+
+  //   return client.pullRequests.getFiles({owner, repo, number})
+  // }
+}
+
+export const HooksCollection = {
+  async one({ self, source, args }) {
+    const { name: owner } = self.match(root.users.one());
+    const { name: repo } = self.match(root.users.one().repos().one());
+    const { id } = args;
+    const result = await octokit.repos.getHook({owner, repo, id});
+    return result.data;
+  },
+
+  async page({ self, source, args }) {
+    const { name: owner } = self.match(root.users.one());
+    const { name: repo } = self.match(root.users.one().repos().one());
+
+    const apiArgs = toApiArgs(args, { owner, repo });
+    const res = await client.repos.getHooks(apiArgs);
+
+    return {
+      items: res.data,
+      next: getNextPageRef(self.page(args), res),
+    };
+  }
+}
+
+export const Hook = {
+  self({ self, parent, source }) {
+    return self || parent.ref.pop().pop().push('one', { id: source.hook_id });
```

Comment:
`self` resolvers should now be written in this form:

`return parent.parent.parent.one({ id: source.hook_id })`

For clarity, here's what it means:

 - `parent` is a ref to the `[Hook]` inside a page
 - `parent.parent` is a ref to the the `HookPage` in the collection
 - `parent.parent.parent` is a ref to the `HookCollection`
 - And from the ref to the collection we just go into the `one` field with the right id